### PR TITLE
Custom Project Name List

### PR DESCRIPTION
**Pull request by iam-rohid**

### CODE REVIEW

File: src/wakatimeScript.ts [hunk 1]
```diff
@@ -40,8 +40,6 @@ chrome.runtime.onMessage.addListener(
         return;
       }
```
Comment (alanhamlett) [inline src/wakatimeScript.ts:42]: Thanks! I forgot to remove this debug code

File: src/core/WakaTimeCore.ts [hunk 2]
```diff
@@ -135,14 +143,18 @@ class WakaTimeCore {
...
     return {
       branch: heartbeat?.branch ?? '<<LAST_BRANCH>>',
       category: heartbeat?.category,
       entity: heartbeat?.entity ?? entity,
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
       id: uuid4(),
       language: heartbeat?.language,
       plugin: heartbeat?.plugin,
-      project: heartbeat?.project ?? '<<LAST_PROJECT>>',
+      project: projectNameFromList ?? heartbeat?.project ?? '<<LAST_PROJECT>>',
```
Comment (alanhamlett) [inline src/core/WakaTimeCore.ts:157]: We still need to use the project detected from DOM, which is in `heartbeat.project`

Comment (alanhamlett) [reply src/core/WakaTimeCore.ts:157]: So it should use defaults ?? in order project list, page, then last project.

Comment (iam-rohid) [reply src/core/WakaTimeCore.ts:157]: If the user has specified a custom project name for a given URL expression, shouldn't we ignore any name from the DOM or URL and use that instead?

Comment (alanhamlett) [reply src/core/WakaTimeCore.ts:157]: Sorry, somehow I didn't see `heartbeat?.project` still used there. All good!

**Pull request merged**